# Support Vector Machine (SVM)

## Importing the libraries

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Importing the dataset

In [2]:
dataset = pd.read_csv('UniversalBank.csv')
X = dataset.iloc[:, 1:-1].values
y = dataset.iloc[:, -1].values

## Splitting the dataset into the Training set and Test set

In [3]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

## Feature Scaling

In [4]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

## Training the SVM model on the Training set

In [5]:
from sklearn.svm import SVC
classifier = SVC(kernel = 'linear', random_state = 0)
classifier.fit(X_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='scale', kernel='linear',
    max_iter=-1, probability=False, random_state=0, shrinking=True, tol=0.001,
    verbose=False)

## Predicting the Test set results

In [6]:
y_pred = classifier.predict(X_test)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[0 0]
 [0 0]
 [0 0]
 ...
 [0 0]
 [0 0]
 [0 0]]


## Making the Confusion Matrix

In [7]:
from sklearn.metrics import confusion_matrix, accuracy_score
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test, y_pred)

[[1134    8]
 [  45   63]]


0.9576

# Conclusion

> In this model I have used data of 5000 customers and the dataset terms are as follows

> data terms:
Dataset Details Attribute 	Description 
* ID: Customer ID
* Age: Customer's age in completed years
* Experience : #years of professional experience
* Income : Annual income of the customer (\$000) 
* ZIPCode : Home Address ZIP code.
* Family : Family size of the customer 
* CCAvg : Avg. spending on credit cards per month (\000) 
* Education : Education Level. 1: Undergrad; 2: Graduate; 3: Advanced/Professional 
* Mortgage : Value of house mortgage if any. (\000) 
* Personal Loan : Did this customer accept the personal loan offered in the last campaign? (Target attribute) 
* Securities Account: Does the customer have a securities account with the bank? 
* CD Account: Does the customer have a certificate of deposit (CD) account with the bank? 
* Online: Does the customer use internet banking facilities? 
* CreditCard: Does the customer use a credit card issued by UniversalBank? 


> Conclusion:
Support Vector Machine (SVM) model has given the accuracy of 95.76%